source/component/PasDoc_Parser.pas

```diff
@@ -2007,25 +2039,20 @@ procedure TParser.ParseType(const U: TPasUnit);
   MethodType: TPasMethod;
   EnumType: TPasEnum;
   T: TToken;
-  IsGeneric: boolean;
 begin
   { Read the type name, preceded by optional "generic" directive.
     Calculate TypeName, IsGeneric, TypeNameWithGeneric.
     FPC requires "generic" directive, but Delphi doesn't,
     so it's just optional for us (serves for some checks later). }
   T := GetNextToken;
   try
-    TypeNameWithGeneric := '';
-    IsGeneric := T.IsStandardDirective(SD_GENERIC);
-    if IsGeneric then
-    begin
-      TypeNameWithGeneric := T.Data + ' ';
-      TypeName := GetAndCheckNextToken(TOK_IDENTIFIER);
-    end else
-    begin
-      CheckToken(T, TOK_IDENTIFIER);
-      TypeName := T.Data;
-    end;
+    if IsGeneric.IsEmpty then
```

Comment:
FPC 3.0.4 doesn't support `IsEmpty` string helper, and (for now) PasDoc supports FPC 3.0.4 compilation still. Please use the traditional comparison `IsGeneric <> ''`.